### ISSUE TITLE
wrong pointer when invalid relation

### ISSUE THREAD

**Issue by happyDemon** [cloudcreativity/laravel-json-api#334]
Let's say I want to create a project, every project should always be assigned a manager and a client (2 relations).
A user doing the input through the interface should select both those relations.

If the user forgot to select select a manager, the validation should return an error with pointer `/data/relationships/manager`. However the pointer I get returned is `/data` since the manager property does not exist within `data.attributes`.

This seems to, strangely, happen in other instances as wel, where the attribute isn't defined in the payload.

Wouldn't it be more logical to always provide a pointer for how the validation rule is defined.
I mean, having a validation rule defined for an attribute, which returns `/data` as a pointer isn't helpful, since the developer has a clear intent for it, which the consumer doesn't know about in this case.

missing attributes: `data/attributes/xx`
missing relations: `data/relationships/xx`

TLDR: There should be a better distinction in what we're validation (difference between relations and attributes) so a correct pointer can be presented in the response.

**Comment by lindyhopchris** [comment]
Hi!

> Wouldn't it be more logical to always provide a pointer for how the validation rule is defined.

The pointer relates to the request JSON, not the rule. As per [the spec](https://tools.ietf.org/html/rfc6901):

> JSON Pointer defines a string syntax for identifying a specific value within a JavaScript Object Notation (JSON) document.

Key phrasing being: **specific value within**

Therefore if a `projects` resource requires a `manager` field to be created, then for this JSON:

```json
{
  "data": {
    "type": "projects",
    "attributes": {
      "title": "My Project"
    }
  }
}
```

the correct pointer is `/data` because the resource object held at the `/data` field is missing a `manager` field (remember that attributes and relationships are collectively called *fields* on a resource object).

Whereas this request JSON:

```json
{
  "data": {
    "type": "projects",
    "attributes": {
      "title": "My Project"
    },
    "relationships": {
      "manager": {
        "data": null
      }
    }
  }
}
```

The pointer should be `/data/relationships/manager`, because the client has provided a `manager` field in their resource object, but a value is required.

TLDR: Send the manager relationship as `null` from the client if the user has not selected a manager.

**Comment by happyDemon** [comment]
I used to use a different implementation for the spec, so these were things I was used to, but apparently wrong 😮. 

Thanks for clarifying everything so quickly, love the work that went into this package by the way 😄
